hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -74,6 +95,123 @@ private boolean hasCryptoAdminKey(final ContractUpdateTransactionBody op) {
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
-        throw new UnsupportedOperationException("Not implemented");
+        final var txn = requireNonNull(context).body();
+        final var op = txn.contractUpdateInstanceOrThrow();
+        final var target = op.contractIDOrThrow();
+
+        final var accountStore = context.readableStore(ReadableAccountStore.class);
+        final var toBeUpdated = accountStore.getContractById(target);
+        validateSemantics(toBeUpdated, context, op);
+        final var changed = update(toBeUpdated, context, op);
+
+        context.serviceApi(TokenServiceApi.class).updateContract(changed);
+        final var recordBuilder = context.recordBuilder(ContractUpdateRecordBuilder.class);
+        recordBuilder.contractID(target);
+    }
+
+    public Account update(
+            @NonNull final Account contract,
+            @NonNull final HandleContext context,
+            @NonNull final ContractUpdateTransactionBody op) {
+        final var builder = contract.copyBuilder();
+        if (op.hasAdminKey()) {
+            if (EMPTY_KEY_LIST.equals(op.adminKey())) {
+                builder.key(contract.key());
+            } else {
+                builder.key(op.adminKey());
+            }
+        }
+        if (op.hasExpirationTime()) {
+            if (contract.expiredAndPendingRemoval()) {
+                builder.expiredAndPendingRemoval(false);
+            }
+            builder.expirationSecond(op.expirationTime().seconds());
+        }
+        if (op.hasAutoRenewPeriod()) {
+            builder.autoRenewSeconds(op.autoRenewPeriod().seconds());
```

Comment:
We should validate `autoRenewPeriod()` also? We have validations for both `expirationTime` and `autoRenewPeriod` in `ExpiryValidatorImpl.resolveUpdateAttempt()` . Can you see if that can be used?

Reply:
I will look into it when I switch back to this PR. Thanks!